addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/SpotifySession.java

```diff
@@ -0,0 +1,1299 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.spotify.internal;
+
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.openhab.binding.spotify.SpotifyBindingConstants;
+import org.openhab.binding.spotify.handler.SpotifyHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * The {@link SpotifySession} maintains an active session with Spotify Web API.
+ * The refreshToken is used to retrieve an accessToken which is used through its validity time.
+ * The class also provides wrappers for various API calls.
+ *
+ * @author Andreas Stenlund - Initial contribution
+ */
+public class SpotifySession implements Runnable {
```

Comment:
I think its not a proper design if the `SpotifySession` is a `Runnable` containing 3 additional executors, can you please refactor that?